src/Microsoft.AspNetCore.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1118,7 +1118,9 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
                     }
                 }
 
-                if (!EndOfFile && CurrentSymbol.Type == HtmlSymbolType.NewLine)
+                if (!EndOfFile &&
+                    CurrentSymbol.Type == HtmlSymbolType.NewLine &&
```

Comment:
Since we're touching this now, lets just change it to `At(HtmlSymbolType.NewLine)`